src/engine/V3Eval.v3

```diff
@@ -1023,6 +1105,34 @@ convert: u64 -> T, extend: (T) -> Tw, convert_to_u: (Tw) -> Uw, f: (Tw, Tw) -> T
 	}
 	return res;
 }
+def do_v128_narrow<Tw, Tn, Un>(a: (u64, u64), b:(u64, u64), src_bits: byte,
+convert: u64 -> Tw, narrow: (Tw) -> Tn, convert_to_u: (Tn) -> Un) -> (u64, u64) {
+	var low = do_v128_narrow0(a, src_bits, convert, narrow, convert_to_u);
+	var high = do_v128_narrow0(b, src_bits, convert, narrow, convert_to_u);
+	return (low, high);
+}
+def do_v128_narrow0<Tw, Tn, Un>(src: (u64, u64), src_bits: byte,
+convert: u64 -> Tw, narrow: (Tw) -> Tn, convert_to_u: (Tn) -> Un) -> u64 {
+	var src_low = src.0, src_high = src.1;
+	var low = 0ul, high = 0ul;
+	var shift = byte.view(src_bits);
+	var res_shift: u6 = 0;
+	var res_shift_inc = u6.view(shift >> 1);
+	for (i = 0; i < 64; i += src_bits) {
+		var val = narrow(convert(u64.view(src_low)));
```

Comment:
I think `src_low` and `src_high` are already `u64`? If so, then the `view` here can be removed.